pmda/parallel.py

```diff
@@ -348,14 +349,17 @@ def run(self,
         if scheduler == 'processes':
             scheduler_kwargs['num_workers'] = n_jobs
 
-        start, stop, step = self._trajectory.check_slice_indices(start,
-                                                                 stop, step)
+        start, stop, step = self._universe.trajectory.check_slice_indices(start,
+                                                                     stop, step)
         n_frames = len(range(start, stop, step))
 
         self.start, self.stop, self.step = start, stop, step
 
         self.n_frames = n_frames
 
+        #  in case _prepare has not set an array.
+        self._results = np.zeros(n_frames)
```

Comment:
Not sure this is always something we want to do. Check the other analysis classes in mda to see if such a default would make sense.

Reply:
This should normally be overridden by the definition inside `_prepare` to suit what `_single_frame` saves.

Reply:
I do not like this reasoning. This means I always have to know about the default and that I can overwrite it in one place. This just looks weird to me. I would rather have to just set this up in once single place.